processing/src/main/java/io/druid/segment/data/GenericIndexed.java

```diff
@@ -407,35 +216,56 @@ public T get(int index)
   @Override
   public int indexOf(T value)
   {
-    return bufferIndexed.indexOf(value);
+    return indexOf(this, value);
+  }
+
+  private int indexOf(Indexed<T> indexed, T value)
```

Comment:
Also, just to be clear "before" means the current master and is the first 2 lines? "after" means with this PR and is the last two lines?  (Sorry, I really am not certain what "before-after" means and so want to clarify that)